octobot_tentacles_manager/managers/tentacle_manager.py

```diff
@@ -0,0 +1,95 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from os import listdir
+from os.path import join, isfile, exists
+from shutil import copyfile, rmtree, copytree
+
+from octobot_commons.logging.logging_util import get_logger
+from octobot_tentacles_manager.constants import TENTACLE_MODULE_FOLDERS, PYTHON_INIT_FILE, \
+    USER_TENTACLE_SPECIFIC_CONFIG_PATH, CONFIG_SCHEMA_EXT, TENTACLE_CONFIG, CONFIG_EXT
+from octobot_tentacles_manager.util.file_util import find_or_create
+
+
+class TentacleManager:
+
+    def __init__(self, tentacle):
+        self.tentacle = tentacle
+        self.target_tentacle_path = None
+
+    async def install_tentacle(self, tentacle_path):
+        self.target_tentacle_path = join(tentacle_path, self.tentacle.tentacle_type.to_path())
+        tentacle_module_path = join(self.target_tentacle_path, self.tentacle.name)
+        await self._update_tentacle_folder(tentacle_path)
+        await self._create_tentacle_init_file_if_necessary(tentacle_module_path)
+        self._import_tentacle_config_if_any(tentacle_module_path)
+
+    async def uninstall_tentacle(self):
+        rmtree(join(self.tentacle.tentacle_path, self.tentacle.name))
```

Comment:
Why not use `UninstallWorker` ?

Reply:
Not use I understand your question but managers are in charge of file system operations, therefore it's from there that a tentacle should be deleted from file system

Reply:
Ok so `UnistallWorker` will use the manager to perform the file deletion ?

Reply:
Yes, TentacleManager is handling the file operations for a given tentacle (create, edit delete)